delivery_state/models/stock_picking.py

```diff
@@ -40,7 +41,34 @@ class StockPicking(models.Model):
     )
 
     def tracking_state_update(self):
-        for picking in self.filtered(lambda p: p.delivery_type):
+        """Call to the service provider API wich should have the method
+        definedin the manner. Can be triggered manually or by cron:
+        <my_provider>_tracking_state_update"""
+        for picking in self.filtered("delivery_type"):
```

Comment:
Avoid to navigate to a related field that is required in carrier.
```suggestion
        for picking in self.filtered("carrier_id"):
```